### PR TITLE
Fix redis authentications - Closes 606

### LINKED ISSUE #606: Do not use cache if Redis authentication failed
Redis initialization always passes a password, even when its null. This causes redis to fail initialization and cause the node to stop functioning.

In order to fix this we need to sanitize the config passed to redis when starting the cache.

In addition, the default passwords should be removed in call cases.

### PR DESCRIPTION
**Pull request by SargeKhan**
closes #606

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -25,9 +25,7 @@
     "redis": {
       "host": "127.0.0.1",
       "port": 6380,
-      "db": 0,
-      "user": "",
-      "password": "vgeuwrKZLYwNyZXNc7vTFbjJ4vQnMGrs"
```
Comment (Isabello) [inline config.json]: Why are we removing this from here? Can we not just delete the entries from the object when null during connect?

Comment (Isabello) [inline config.json]: We 100% have to be flexible and offer users this option.

Comment (SargeKhan) [inline config.json]: I get what you mean now. Making the required changes.

File: helpers/cache.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 
 var redis = require('redis');
+var _ = require('lodash');
```
Comment (Isabello) [inline helpers/cache.js]: Find a way to do it without lodash please.

File: config.json [hunk 3]
```diff
@@ -23,11 +23,10 @@
...
     "redis": {
-      "host": "127.0.0.1",
-      "port": 6380,
-      "db": 0,
-      "user": "",
-      "password": "vgeuwrKZLYwNyZXNc7vTFbjJ4vQnMGrs"
+        "host": "127.0.0.1",
+        "port": 6380,
+        "db": 0,
+        "password": null
```
Comment (Isabello) [inline config.json]: where did the user field go?

why is the password null instead of an empty string?

Comment (SargeKhan) [inline config.json]: went through the docs again, user field doesn't mean anything for redis.
Empty string is a valid password in redis. System can have redis configuration which has `requirepass ""`. So, we need to differentiate b/w two.

Comment (Isabello) [inline config.json]: ok user field is ok to go.

are you telling me redis interprets `""` as a password?

Comment (SargeKhan) [inline config.json]: Yes, that's what i mean.

Comment (Isabello) [inline config.json]: thats super poor design on their part. Ok this is fine I will refactor lisk build to accomodate this change.

**Comment by Isabello** [comment]
Please push your branch upstream to HQ and open a new PR from there so that it can be built in Jenkins.

**Pull request closed without merging**